src/test/server/server_test_runner.cpp

```diff
@@ -221,13 +221,79 @@ TEST_F(ServerTestRunner, TestParallelConnections) {
     thread_futures.emplace_back(std::async(std::launch::async, connection_run));
   }
 
-  for (auto& thread_fut : thread_futures) {
+  for (auto& thread_future : thread_futures) {
     // We give this a lot of time, not because we need that long for 100 threads to finish, but because sanitizers and
     // other tools like valgrind sometimes bring a high overhead that exceeds 10 seconds.
-    if (thread_fut.wait_for(std::chrono::seconds(150)) == std::future_status::timeout) {
+    if (thread_future.wait_for(std::chrono::seconds(150)) == std::future_status::timeout) {
       ASSERT_TRUE(false) << "At least one thread got stuck and did not commit.";
+      thread_future.get();
```

Comment:
What's the purpose of this line?

Reply:
Added "Retrieve the future so that exceptions stored in its state are thrown"

Reply:
Please refer to @toniSta's comment from https://github.com/hyrise/hyrise/issues/1908

Reply:
I'd rather have the comments self-contained where possible. Which information are you missing?